modules/caddyhttp/reverseproxy/reverseproxy.go

```diff
@@ -730,10 +762,18 @@ func (h *Handler) reverseProxy(rw http.ResponseWriter, req *http.Request, origRe
 	server := req.Context().Value(caddyhttp.ServerCtxKey).(*caddyhttp.Server)
 	shouldLogCredentials := server.Logs != nil && server.Logs.ShouldLogCredentials
 
+	// Default to using the transport configured during provisioning stage
+	transport := h.Transport
+
+	// If we have a HTTP transport, try to replace the TLS servername
+	if tmpTransport, ok := transport.(*HTTPTransport); ok {
+		transport = h.replaceTLSServername(tmpTransport, repl)
+	}
```

Comment:
Instead of using a type assertion here, can this be done in `RoundTrip()` directly? Seeing as the transport is short-lived and all.